ethyca/fides
# Export data map from server resources instead of manifest dir

Closes #659

### Code Changes

* [x] Include an `organization_fides_key` when exporting a data map
* [x] Filter resources by `organization_fides_key`
* [x] Remove requirement to provide a manifest directory (currently used as output location still)
* [x] Pull all resources from the server instead of just what is listed in the manifest directory
* [x] Updated guide to a data map with option to use `demo_resources/`

### Steps to Confirm

* [x] Ensure exporting a data map contains all resources from both `.fides/` and `demo_resources/` after applying them to the server
* [x] Ensure items from a separate organization are filtered off of an exported data map

### Pre-Merge Checklist

* Documentation Updated:
  - [x] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [x] documentation issue created (tag docs-team to complete issue separately)

### Description Of Changes

Previously only fides keys found in a specified manifest directory were included in a data map. This not only led to a bug when an organization is not defined in a manifest directory but also is impractical when working inside a large organization.

This change will have a data map exported for a specific organization, which is more aligned with our documentation around organizations and expectations of users.

Still looking at the arguments used here and seeing if an option is a valid replacement, either for output file or `organization_fides_key`


Updated steps for validation:
1. `nox -s cli`
2. Apply both sets of resources to the server: `fidesctl apply && fidesctl apply demo_resources/`
3. Export the combined data map: `fidesctl export datamap`
4. Validate the exported data map in the `.fides/` directory contains three systems
5. Change the output directory: `fidesctl export datamap demo_resources/`

Filtering test (slightly more complicated)
6. Create a basic organization in the `.fides/` directory with a different name (i.e. `new_organization`)
7. Apply the new `organization_fides_key` to the other resources in the `.fides/` directory
8. `fidesctl export datamap demo_resources/`
9. Validate the resources from the `.fides/` directory are no longer in the exported datamap

**Some open questions:**
* Currently, the other resources to export still point to a manifest directory - should we change this as well?
* The command itself feels a bit chunky, `organization_fides_key` is now an option versus an argument but opinions welcome
